toolkit/scripts/pkggen.mk

```diff
@@ -89,13 +89,40 @@ $(specs_file): $(chroot_worker) $(BUILD_SPECS_DIR) $(build_specs) $(build_spec_d
 		$(if $(TARGET_ARCH),--target-arch="$(TARGET_ARCH)") \
 	        --output $@
 
+grapher_cycle_flags :=
+
+grapher_extra_flags :=
+ifeq ($(DISABLE_UPSTREAM_REPOS),y)
+grapher_extra_flags += --disable-upstream-repos
+endif
+
+ifeq ($(USE_PREVIEW_REPO),y)
+grapher_extra_flags += --use-preview-repo
+endif
+
+ifeq ($(RESOLVE_CYCLES_FROM_REPO),y)
+grapher_cycle_flags += --resolve-cycles-from-repo
+endif
+
+grapher_cycle_flags += --output-dir=$(CACHED_RPMS_DIR)/cache
+grapher_cycle_flags += --rpm-dir=$(RPMS_DIR)
+grapher_cycle_flags += --toolchain-rpms-dir=$(TOOLCHAIN_RPMS_DIR)
+grapher_cycle_flags += --toolchain-manifest=$(TOOLCHAIN_MANIFEST)
+grapher_cycle_flags += --tmp-dir=$(cache_working_dir)
+grapher_cycle_flags += --tdnf-worker=$(chroot_worker)
```

Comment:
With this flag, you'll need the grapher to now also depend on `$(chroot_worker)`, so that it's available before `make` tries to run it.

Please check the dependencies for other flags as well.

Reply:
I'm not sure, I understood. I agree that the `$(chroot_worker)` is required for the grapher now but that's exactly why I mentioned we should add it to the list of dependencies for `$(graph_file)` right next to `$(specs_file) $(go-grapher) $(toolchain_rpms)` that's already there.

Reply:
Ohh, Got it! Added the dependencies. Please check